ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/KsqlKeyExtractor.java

```diff
@@ -15,30 +15,43 @@
 
 package io.confluent.ksql.execution.streams;
 
-import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
 
 import io.confluent.ksql.GenericKey;
 import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.execution.transform.ExpressionEvaluator;
+import io.confluent.ksql.logging.processing.ProcessingLogger;
 import java.util.Objects;
 import java.util.function.Function;
+import java.util.function.Supplier;
 
 public final class KsqlKeyExtractor<KRightT> implements Function<GenericRow, KRightT> {
 
-  private final int leftJoinColumnIndex;
+  private final ExpressionEvaluator expressionEvaluator;
+  private final ProcessingLogger  processingLogger;
+  private final Supplier<String> errorMessage;
 
-  KsqlKeyExtractor(final int leftJoinColumnIndex) {
-    checkArgument(
-        leftJoinColumnIndex >= 0,
-        "leftJoinColumnIndex negative: " + leftJoinColumnIndex
+  KsqlKeyExtractor(final ExpressionEvaluator expressionEvaluator,
+                   final ProcessingLogger processingLogger) {
+    this.expressionEvaluator = requireNonNull(expressionEvaluator);
+    this.processingLogger = processingLogger;
```

Comment:
Any reason we don't require this to be non-null?

Reply:
We only "forward" it to `expressionEvaluator.evaluate()` but don't use it ourselves, so I though we don't need this (make testing simpler :))